Source/Assets/Scripts/PlayerInteractor.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+
+public class PlayerInteractor : MonoBehaviour
+{ 
+    
+    private static PlayerInteractor _instance;
+    
+   [Header("Interaction")] 
+   [SerializeField] private float interactionDistance = 4f;
+   [SerializeField] private LayerMask interactableLayerMask;
+   [SerializeField] private GameObject interactionUI;
+   [SerializeField] private TextMeshProUGUI interactionText;
+   private Interactable interactable;
+   
+   [Space]
+   [Header("Inspection")]
+   public float inspectionDistance = 1f;
+   public bool isInspecting = false;
+
+   [SerializeField] private Interactable currentlyInspected;
+   private Vector2 objectRotation = Vector2.zero;
+   
+   public static PlayerInteractor Instance
+   {
+       get
+       {
+           if (_instance == null) _instance = GameObject.FindObjectOfType<PlayerInteractor>();
+           return _instance;
+       }
+   }
+
+   
+   private void Update()
+   {
+       if (isInspecting)
+       {
+           interactionUI.SetActive(false);
```

Comment:
For UI, see if you can use Canvas.enabled instead of SetActive on the game object itself